### PR TITLE
feat: pretty print Quoted token stream

### LINKED ISSUE #6110: The output of printing quoted values is hard to understand
### Problem

In comptime code when we `println(q)` and `q` is `Quoted`, the output is just the list of tokens separated by spaces. One usually outputs these to be able to undersand and debug what's going on, but because the code is not somehow formatted it's really hard to do. What one (at least I) end up doing is copying that output to a separate file and running `nargo fmt` on it (well, I let the editor do it for me) to understand things.

It would be nice if the output of that stream of tokens was somehow easier to understand.

### Happy Case

The output of printing quoted values is easier to understand.

### Workaround

None

### Workaround Description

_No response_

### Additional Context

_No response_

### Project Impact

None

### Blocker Context

_No response_

### Would you like to submit a PR for this Issue?

None

### Support Needs

_No response_

### PR DESCRIPTION
**Pull request by asterite**
# Description

## Problem

Resolves #6110

## Summary

This should make it much easier to debug macro code.

For example, compare this old output:

```noir
quote { pub fn public_entrypoint ( selector : Field ) { let cdoffset = 0 ; let selector = dep :: aztec :: protocol_types :: abis :: function_selector :: FunctionSelector :: from_field ( selector ) ; if selector == (FunctionSelector { inner: 3043802001 }) { let input_calldata : [ Field ; 3 ] = calldata_copy ( 0 , 3 ) ; let input_calldata_index = 0 ; let ( inputs , input_calldata_index ) = PublicContextInputs :: from_calldata ( input_calldata , input_calldata_index ) ; let ( a , input_calldata_index ) = Field :: from_calldata ( input_calldata , input_calldata_index ) ; let return_value = set_storage_single ( inputs , a ) ; let mut return_array : [ Field ; 1 ] = [ 0 ; 1 ] ; return_value . to_calldata ( & mut return_array , 0 ) ; avm_return ( return_array ) ; } else if selector == (FunctionSelector { inner: 1695189557 }) { let input_calldata : [ Field ; 2 ] = calldata_copy ( 0 , 2 ) ; let input_calldata_index = 0 ; let ( inputs , input_calldata_index ) = PublicContextInputs :: from_calldata ( input_calldata , input_calldata_index ) ; let return_value = read_storage_single ( inputs ) ; let mut return_array : [ Field ; 1 ] = [ 0 ; 1 ] ; return_value . to_calldata ( & mut return_array , 0 ) ; avm_return ( return_array ) ; } else if selector == (FunctionSelector { inner: 1009316778 }) { let input_calldata : [ Field ; 3 ] = calldata_copy ( 0 , 3 ) ; let input_calldata_index = 0 ; let ( inputs , input_calldata_index ) = PublicContextInputs :: from_calldata ( input_calldata , input_calldata_index ) ; let ( a , input_calldata_index ) = Field :: from_calldata ( input_calldata , input_calldata_index ) ; let return_value = set_read_storage_single ( inputs , a ) ; let mut return_array : [ Field ; 1 ] = [ 0 ; 1 ] ; return_value . to_calldata ( & mut return_array , 0 ) ; avm_return ( return_array ) ; } else { panic ( f"Unknown selector" ) } } }
```

to this new one:

```noir
quote {
    pub fn public_entrypoint(selector: Field) {
        let cdoffset = 0;
        let selector = dep::aztec::protocol_types::abis::function_selector::FunctionSelector::from_field(selector);
        if selector == (FunctionSelector { inner: 1009316778 }) {
            let input_calldata: [Field; 3] = calldata_copy(0, 3);
            let input_calldata_index = 0;
            let(inputs, input_calldata_index) = PublicContextInputs::from_calldata(input_calldata, input_calldata_index);
            let(a, input_calldata_index) = Field::from_calldata(input_calldata, input_calldata_index);
            let return_value = set_read_storage_single(inputs, a);
            let mut return_array: [Field; 1] = [0; 1];
            return_value.to_calldata( & mut return_array, 0);
            avm_return(return_array);
        } else if selector == (FunctionSelector { inner: 3043802001 }) {
            let input_calldata: [Field; 3] = calldata_copy(0, 3);
            let input_calldata_index = 0;
            let(inputs, input_calldata_index) = PublicContextInputs::from_calldata(input_calldata, input_calldata_index);
            let(a, input_calldata_index) = Field::from_calldata(input_calldata, input_calldata_index);
            let return_value = set_storage_single(inputs, a);
            let mut return_array: [Field; 1] = [0; 1];
            return_value.to_calldata( & mut return_array, 0);
            avm_return(return_array);
        } else if selector == (FunctionSelector { inner: 1695189557 }) {
            let input_calldata: [Field; 2] = calldata_copy(0, 2);
            let input_calldata_index = 0;
            let(inputs, input_calldata_index) = PublicContextInputs::from_calldata(input_calldata, input_calldata_index);
            let return_value = read_storage_single(inputs);
            let mut return_array: [Field; 1] = [0; 1];
            return_value.to_calldata( & mut return_array, 0);
            avm_return(return_array);
        } else {
            panic(f"Unknown selector")
        }
    }
}
```

## Documentation

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: compiler/noirc_frontend/src/hir/comptime/value.rs [hunk 1]
```diff
@@ -761,6 +755,255 @@ impl<'token, 'interner> Display for TokenPrinter<'token, 'interner> {
     }
 }
 
+fn print_quoted(
+    tokens: &[Token],
+    indent: usize,
+    interner: &NodeInterner,
+    f: &mut std::fmt::Formatter<'_>,
+) -> std::fmt::Result {
```
Comment (jfecher) [inline compiler/noirc_frontend/src/hir/comptime/value.rs:763]: There's getting to be a lot of value formatting code. Can you move it to it's own sub-module in a new file?

File: compiler/noirc_frontend/src/hir/comptime/quoted_pretty_printing.rs
Comment (jfecher) [inline compiler/noirc_frontend/src/hir/comptime/quoted_pretty_printing.rs:1]: Could you add _all_ pretty printing functions for values to this new module? May need to rename it as a result.

E.g. the formatting function which takes the interner, the remove_quoted_in_* functions as well

Comment (asterite) [reply compiler/noirc_frontend/src/hir/comptime/quoted_pretty_printing.rs:1]: All good! These refactors don't take that much time :-)

Comment (asterite) [reply compiler/noirc_frontend/src/hir/comptime/quoted_pretty_printing.rs:1]: Done!

**Review by jfecher — COMMENTED** [review]
Tested locally and formatting seems good. Can we apply this to `InterpreterError::FailedToParseMacro` as well?

**Comment by asterite** [comment]
> Can we apply this to InterpreterError::FailedToParseMacro as well?

I think so! The only "problem" is that we don't have a `NodeInterner` when we display the tokens, but we could change `FailedToParseMacro` to take the string instead of the tokens. What do you think?

**Comment by jfecher** [comment]
>  to take the string instead of the tokens. What do you think?

Hmm, yeah that'd be better. As long as the formatting improves I think it is fine to swap out the types used in error variants

**Review by jfecher — APPROVED** [review]
LGTM

**Pull request merged**